tests/integration/route_test.py

```diff
@@ -807,6 +807,130 @@ def test_add_and_remove_ipv4_link_local_route(eth1_static_gateway_dns):
     assert IPV4_ADDRESS2 not in routes_output
 
 
+@pytest.mark.tier1
+def test_add_route_with_cwnd(eth1_up):
+    routes = [
+        {
+            Route.NEXT_HOP_INTERFACE: "eth1",
+            Route.DESTINATION: IPV4_TEST_NET1,
+            Route.NEXT_HOP_ADDRESS: IPV4_ADDRESS1,
+            Route.CWND: 20,
+        },
+        {
+            Route.NEXT_HOP_INTERFACE: "eth1",
+            Route.DESTINATION: IPV6_TEST_NET1,
+            Route.NEXT_HOP_ADDRESS: IPV6_GATEWAY1,
+            Route.CWND: 20,
+        },
+    ]
+    libnmstate.apply(
+        {
+            Interface.KEY: [ETH1_INTERFACE_STATE],
+            Route.KEY: {Route.CONFIG: routes},
+        }
+    )
+    cur_state = libnmstate.show()
+    assert_routes(routes, cur_state)
+
+
+@pytest.mark.tier1
+def test_delete_route_with_cwnd(eth1_up):
+    libnmstate.apply(
+        {
+            Interface.KEY: [ETH1_INTERFACE_STATE],
+            Route.KEY: {
+                Route.CONFIG: [
+                    {
+                        Route.NEXT_HOP_INTERFACE: "eth1",
+                        Route.DESTINATION: IPV4_TEST_NET1,
+                        Route.NEXT_HOP_ADDRESS: IPV4_ADDRESS1,
+                        Route.CWND: 20,
+                    },
+                    {
+                        Route.NEXT_HOP_INTERFACE: "eth1",
+                        Route.DESTINATION: IPV6_TEST_NET1,
+                        Route.NEXT_HOP_ADDRESS: IPV6_GATEWAY1,
+                        Route.CWND: 20,
+                    },
+                ]
+            },
+        }
+    )
+
+    absent_routes = [{Route.CWND: 20, Route.STATE: Route.STATE_ABSENT}]
+    libnmstate.apply({Route.KEY: {Route.CONFIG: absent_routes}})
+
+    expected_routes = []
+    cur_state = libnmstate.show()
+    assert_routes(expected_routes, cur_state)
```

Comment:
The `assert_routes()` does nothing against a empty array. You need to check the routes are removed manually.